File: JAVA_SIS104/app/src/main/java/com/example/prueba_java/backend/grafico.java
package com.example.prueba_java.backend;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.view.View;

import androidx.annotation.NonNull;

public class grafico extends View {
    public grafico(Context context){
        super(context);
    }

    @Override
    protected void onDraw(@NonNull Canvas canvas) {
        super.onDraw(canvas);
        canvas.drawColor(Color.BLUE);

        Paint paint = new Paint();
        paint.setColor(Color.WHITE);
        canvas.drawLine(10, 10, 50, 50, paint);
        for(int i=1; i<=500; i+=1){
            paint.setColor(Color.argb(150, 120, 60, 1));
            canvas.drawLine(10, 10, 510-i, 10+i, paint);
        }

        Paint paint3 = new Paint();
        paint3.setColor(Color.GRAY);
        canvas.drawCircle(500, 1400, 150, paint3);

        Paint paint2 = new Paint();
        paint2.setColor(Color.GREEN);
        canvas.drawCircle(700, 1500, 150, paint2);

        Paint paintLine = new Paint();
        paintLine.setColor(Color.WHITE);
        paintLine.setStrokeWidth(10);

        canvas.drawLine(500 - 150, 1400, 700 - 150, 1500, paintLine);
        canvas.drawLine(500 + 150, 1400, 700 + 150, 1500, paintLine);

    }
}

File: KOTLIN_SIS104/app/src/main/java/com/example/prueba_kotlin/backend/Grafico.kt
package com.micky.primerkotlin.backend

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.Color.WHITE
import android.util.DisplayMetrics
import android.view.View
import android.view.WindowManager

@Suppress("DEPRECATION")
class Grafico(context: Context?) : View(context) {

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        canvas.drawColor(WHITE)
        var paint = Paint()
        paint.color = Color.BLUE
        canvas.drawCircle(100.0f, 100.0f, 50.0f, paint)

        val escala = 0.00001

        val tamanoPantalla = TPantalla()
        val ancho = tamanoPantalla.first
        val alto = tamanoPantalla.second

        paint.color = Color.BLACK
        paint.textSize = 50.0f
        canvas.drawText("Ancho: $ancho px, Alto: $alto px", 10.0f, 200.0f, paint)
        canvas.drawText("Factor de escala: $escala px", 10.0f, 250.0f, paint)

        val tamañoCuadradoAbsoluto = 500.0f
        val tamañoCuadrado = if (tamañoCuadradoAbsoluto > ancho) {
            ancho*tamañoCuadradoAbsoluto* escala.toFloat()
        } else {
            tamañoCuadradoAbsoluto

        }
        canvas.drawText("tamaño: $tamañoCuadrado cuadrado px", 10.0f, 300.0f, paint)
        paint.color = Color.RED
        val xCuadrado = (ancho - tamañoCuadrado) / 2.0f
        val yCuadrado = 200.0f
        canvas.drawRect(xCuadrado, yCuadrado, xCuadrado + tamañoCuadrado, yCuadrado + tamañoCuadrado, paint)

    }

    fun TPantalla(): Pair<Int, Int> {
        val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        val displayMetrics = DisplayMetrics()

        // Obtener las métricas de la pantalla actual
        windowManager.defaultDisplay.getMetrics(displayMetrics)

        // Obtener el ancho y alto de la pantalla en píxeles
        val ancho = displayMetrics.widthPixels
        val alto = displayMetrics.heightPixels

        return Pair(ancho, alto)
    }
}
